src/harvest.cpp

```diff
@@ -43,20 +43,23 @@ bool string_id<harvest_drop_type>::is_valid() const
 
 translation harvest_drop_type::field_dress_msg( bool succeeded ) const
 {
-    return SNIPPET.random_from_category( succeeded ? msg_fielddress_success :
-                                         msg_fielddress_fail ).value_or( translation() );
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
+    return SNIPPET.random_from_category(
+               succeeded ? msg_fielddress_success : msg_fielddress_fail ).value_or( translation() );
 }
 
 translation harvest_drop_type::butcher_msg( bool succeeded ) const
 {
-    return SNIPPET.random_from_category( succeeded ? msg_butcher_success : msg_butcher_fail ).value_or(
-               translation() );
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
+    return SNIPPET.random_from_category(
+               succeeded ? msg_butcher_success : msg_butcher_fail ).value_or( translation() );
 }
 
 translation harvest_drop_type::dissect_msg( bool succeeded ) const
 {
-    return SNIPPET.random_from_category( succeeded ? msg_dissect_success : msg_dissect_fail ).value_or(
-               translation() );
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
```

Comment:
?

Reply:
Something about the implementation of `unique_ptr` has changed in C++17 which causes what I believe to be a clang-tidy false-positive here.  I tried some tweaks to fix it, but nothing worked, so in the end I chose to suppress it.

Part of me is a little worried that it's not a false-positive and there really is a potential memory leak here, but the `value_ptr` implementation looks pretty solid to me...

Reply:
In case you're curious to analyse it yourself, here's the error message we get without the suppression:
```
.../Cataclysm-DDA/src/harvest.cpp:46:5: error: Potential leak of memory pointed to by field '_M_head_impl' [clang-analyzer-cplusplus.NewDeleteLeaks,-warnings-as-errors]
    return SNIPPET.random_from_category(
    ^
.../Cataclysm-DDA/src/harvest.cpp:47:16: note: Assuming 'succeeded' is false
               succeeded ? msg_fielddress_success : msg_fielddress_fail ).value_or( translation() );
               ^
.../Cataclysm-DDA/src/harvest.cpp:47:16: note: '?' condition is false
.../Cataclysm-DDA/src/harvest.cpp:46:12: note: Calling 'optional::value_or'
    return SNIPPET.random_from_category(
           ^
.../Cataclysm-DDA/src/optional.h:146:20: note: Assuming field 'full' is true
            return full ? get() : static_cast<T>( other );
                   ^
.../Cataclysm-DDA/src/optional.h:146:20: note: '?' condition is true
.../Cataclysm-DDA/src/optional.h:146:27: note: Calling implicit copy constructor for 'translation'
            return full ? get() : static_cast<T>( other );
                          ^
.../Cataclysm-DDA/src/optional.h:146:27: note: Calling copy constructor for 'value_ptr<std::basic_string<char>>'
.../Cataclysm-DDA/src/value_ptr.h:27:33: note: '?' condition is true
            std::unique_ptr<T>( other ? new T( *other ) : nullptr ) {}
                                ^
.../Cataclysm-DDA/src/value_ptr.h:27:41: note: Memory is allocated
            std::unique_ptr<T>( other ? new T( *other ) : nullptr ) {}
                                        ^
.../Cataclysm-DDA/src/optional.h:146:27: note: Returning from copy constructor for 'value_ptr<std::basic_string<char>>'
            return full ? get() : static_cast<T>( other );
                          ^
.../Cataclysm-DDA/src/optional.h:146:27: note: Returning from copy constructor for 'translation'
.../Cataclysm-DDA/src/harvest.cpp:46:12: note: Returned allocated memory
    return SNIPPET.random_from_category(
           ^
.../Cataclysm-DDA/src/harvest.cpp:46:5: note: Potential leak of memory pointed to by field '_M_head_impl'
    return SNIPPET.random_from_category(
    ^
```

Reply:
Looks similar to the type of false positive in https://bugs.llvm.org/show_bug.cgi?id=38176. Switching to `libc++` may help.

Reply:
It might help, but I'd prefer not to do that, especially not in this PR.  I'd like to keep what clang-tidy is analyzing as close as possible to what we build in the rest of CI.  And I don't think we use libc++ except on Mac.

Longer term, if we add the clang-tidy check that forbids any use of raw new or delete, then we can probably just disable this check.

Reply:
@jbytheway if you resolve this then someone can merge this ;)